File: frontend/src/app/pages/raporlar/oda-musaitlik/oda-musaitlik-rapor.dto.ts
export interface OdaMusaitlikOzetDto {
    toplamOdaSayisi: number;
    tamamenBosOdaSayisi: number;
    tamamenDoluOdaSayisi: number;
    kismenMusaitOdaSayisi: number;

    toplamGunSayisi: number;
    toplamOdaGunSayisi: number;
    bosOdaGunSayisi: number;
    doluOdaGunSayisi: number;
    musaitlikOrani: number;
}

export interface OdaMusaitlikGunDto {
    tarih: string;
    gunAdi: string;
    bosMu: boolean;
    doluMu: boolean;
    rezervasyonId: number | null;
    referansNo: string | null;
    misafirAdiSoyadi: string | null;
    rezervasyonDurumu: string | null;
    rezervasyonDurumuLabel: string | null;
}

export interface OdaMusaitlikOdaDto {
    odaId: number;
    odaNo: string;
    binaAdi: string | null;
    odaTipiAdi: string | null;
    kapasite: number;

    musaitlikDurumu: string;
    musaitlikDurumuLabel: string;

    toplamGunSayisi: number;
    bosGunSayisi: number;
    doluGunSayisi: number;
    musaitlikOrani: number;

    gunler: OdaMusaitlikGunDto[];
}

export interface OdaMusaitlikRaporDto {
    tesisId: number;
    tesisAdi: string | null;
    baslangic: string;
    bitis: string;
    durum: string;
    odaTipiId: number | null;
    odaTipiAdi: string | null;
    kapasite: number | null;
    baslik: string;

    ozet: OdaMusaitlikOzetDto;
    odalar: OdaMusaitlikOdaDto[];
}

File: backend/Raporlar/OdaMusaitlik/Dto/OdaMusaitlikRaporDto.cs
namespace STYS.Raporlar.OdaMusaitlik.Dto;

public class OdaMusaitlikRaporDto
{
    public int TesisId { get; set; }
    public string? TesisAdi { get; set; }
    public DateTime Baslangic { get; set; }
    public DateTime Bitis { get; set; }
    public string Durum { get; set; } = "tumu";
    public int? OdaTipiId { get; set; }
    public string? OdaTipiAdi { get; set; }
    public int? Kapasite { get; set; }
    public string Baslik { get; set; } = "";

    public OdaMusaitlikOzetDto Ozet { get; set; } = new();
    public List<OdaMusaitlikOdaDto> Odalar { get; set; } = [];
}

public class OdaMusaitlikOzetDto
{
    public int ToplamOdaSayisi { get; set; }
    public int TamamenBosOdaSayisi { get; set; }
    public int TamamenDoluOdaSayisi { get; set; }
    public int KismenMusaitOdaSayisi { get; set; }

    public int ToplamGunSayisi { get; set; }
    public int ToplamOdaGunSayisi { get; set; }
    public int BosOdaGunSayisi { get; set; }
    public int DoluOdaGunSayisi { get; set; }
    public decimal MusaitlikOrani { get; set; }
}

public class OdaMusaitlikOdaDto
{
    public int OdaId { get; set; }
    public string OdaNo { get; set; } = "";
    public string? BinaAdi { get; set; }
    public string? OdaTipiAdi { get; set; }
    public int Kapasite { get; set; }

    public string MusaitlikDurumu { get; set; } = "";
    public string MusaitlikDurumuLabel { get; set; } = "";

    public int ToplamGunSayisi { get; set; }
    public int BosGunSayisi { get; set; }
    public int DoluGunSayisi { get; set; }
    public decimal MusaitlikOrani { get; set; }

    public List<OdaMusaitlikGunDto> Gunler { get; set; } = [];
}

public class OdaMusaitlikGunDto
{
    public DateTime Tarih { get; set; }
    public string GunAdi { get; set; } = "";
    public bool BosMu { get; set; }
    public bool DoluMu { get; set; }
    public int? RezervasyonId { get; set; }
    public string? ReferansNo { get; set; }
    public string? MisafirAdiSoyadi { get; set; }
    public string? RezervasyonDurumu { get; set; }
    public string? RezervasyonDurumuLabel { get; set; }
}
